### PR TITLE
Create, Change and Update Groups

### LINKED ISSUE #22: Page d'accueil d'un groupe
### Description

En tant qu'utilisateur je souhaite pouvoir naviguer sur la page d'un groupe, créer et rejoindre un groupe.

### Definition of Done

- Une page "Home" existe et affiche le nom du groupe actuel
  - Le groupe actuel de l'utilisateur est stockée dans le local storage du navigateur 
- Un dropdown menu existe pour changer de groupe
  - Affiche tous les groupes dont l'utilisateur fait parti
  - Affiche les groupes dans lequel l'utilisateur à été invité
  - Un bouton permet de créer un groupe
- Si l'utilisateur ne fait parti d'aucun groupe, un empty state lui permet d'en créer un

### Estimation

Estimée à **3** JH.

### PR DESCRIPTION
**Pull request by dhbrojas**
#### Description

Add the ability to create a group on the front-end, switch between multiple groups and update the name and description.

Fixes #22 

#### Changelog

- [ENHANCEMENT] Add a `GroupContext` to manage the currently active group.
- [ENHANCEMENT] Add a mechanism for detecting when the selected group has been deleted or when the user has been removed from said group.
- [ENHANCEMENT] Add an empty state for when the user doesn't have any groups yet.
- [ENHANCEMENT] Add a dropdown to switch groups and create new ones.
- [ENHANCEMENT] Add a tab menu in the `GroupView` to switch between Notes, Settings and Upgrade menus.
- [ENHANCEMENT] Add fetching utilities to communicate with the backend.
- [ENHANCEMENT] Add caching and invalidation on mutation for requests to keep the state consistent across the app.

### CODE REVIEW

**Review by gabriel654165 — COMMENTED** [review]
It looks good for me, but there is a lot a line that I don't understand well. Maybe reviews of front developers will be more relevant

**Comment by dhbrojas** [comment]
> It looks good for me, but there is a lot a line that I don't understand well. Maybe reviews of front developers will be more relevant

Okay thanks @gabriel654165 ! Next time let me know which ones are confusing because I admit the code is pretty cluttered in some places so I'll try to improve the sections you have trouble understanding!

File: src/panels/GroupOverviewPanel.tsx [hunk 1]
```diff
@@ -1,10 +1,33 @@
+import { ChevronRightIcon } from '@heroicons/react/24/outline'
 import React from 'react'
 import PanelSkeleton from '../components/view/PanelSkeleton'
 
+const MembersMenu: React.FC = () => {
+  const members = [
+    {id: '1234', role: 'admin', email: 'edouard@epitech.com', name: 'Édouard', online: true, lastSeen: '2 hours ago', avatarUrl: 'https://avatars.githubusercontent.com/u/58398928?s=40&u=23c492131e2aaeebd06777874bd1a1ab950cdd7c&v=4'},
+    {id: '1234', role: null, email: 'maxime@epitech.com', name: 'Maxime', online: false, lastSeen: '2 hours ago', avatarUrl: 'https://avatars.githubusercontent.com/u/61683870?s=40&u=14e2d9297bed88279fcba593f1ecd0d8ee728196&v=4'},
+    {id: '1234', role: null, email: 'thomas@epitech.com', name: 'Thomas', online: false, lastSeen: 'Yesterday', avatarUrl: 'https://avatars.githubusercontent.com/u/58695101?s=40&u=0ab42103368360947951de967dd4ffdcb8b0ee5a&v=4'},
+  ]
```
Comment (kfleury) [inline src/panels/GroupOverviewPanel.tsx:10]: tu laisse en dur ?

File: src/hooks/click.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  React.useEffect(() => {
+    function handleClickOutside(event) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        cb()
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
```
Comment (kfleury) [inline src/hooks/click.ts:13]: les event listener sont bien remove plus tard ?

File: src/hooks/api/groups.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+export const useGetGroup = newQueryHook<GetGroupRequest, GetGroupResponse>(
+  (req) => `groups/${req.group_id}`,
+  ['group_id']
+)
+
+export const useGetCurrentGroup = (params?: QueryHookParams) => {
+  const groupContext = useGroupContext()
+
+
```
Comment (kfleury) [inline src/hooks/api/groups.ts:18]: une raison particulière à mettre 2 return ?

File: src/contexts/group.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+};
+
+export const GroupContext = React.createContext<TGroupContext | undefined>(undefined)
+
+// Manage unauthenticated user sessions. This context is only available whithin
+// unauthenticated views.
+export const useGroupContext = () => {
+  const context = React.useContext(GroupContext)
+  if (context === undefined) {
+    throw new Error('GroupContext used outside of provider')
```
Comment (kfleury) [inline src/contexts/group.ts:15]: le throw est bien catch plus tard ?

File: src/hooks/click.ts (hunk 2 above)
Comment (dhbrojas) [reply src/hooks/click.ts:13]: Oui! Le call à removeEventListener est appelé quand le component est unmount.

File: src/contexts/group.ts (hunk 4 above)
Comment (dhbrojas) [reply src/contexts/group.ts:15]: Oui et non. Si il y a une error boundary dans le component tree, l'erreur sera catch. Sinon l'application "crash" mais ce n'est pas un problème car si ça advient c'est la faute du programmeur et ce sera catch instantly quand le code sera run.

File: src/hooks/api/groups.ts (hunk 3 above)
Comment (dhbrojas) [reply src/hooks/api/groups.ts:18]: Nope! My bad, will fix now or next time.

File: src/panels/GroupOverviewPanel.tsx (hunk 1 above)
Comment (dhbrojas) [reply src/panels/GroupOverviewPanel.tsx:10]: "Oui". Ce component est appellé nulle part pour l'instant et en fait c'est temporaire, je vais modifier ça quand je ferais les invites et les membres que je changerai. Ce fichier aurait pas du être inclus dans la PR. My bad.

**Pull request merged**